ui/page/wallets/wallet_page.go

```diff
@@ -1079,6 +1085,61 @@ func (pg *WalletPage) deleteBadWallet(badWalletID int) {
 		}).Show()
 }
 
+func (pg *WalletPage) listenForTxNotifications() {
+	go func() {
+		for {
+			var notification interface{}
+
+			select {
+			case notification = <-pg.Receiver.NotificationsUpdate:
+			case <-pg.ctx.Done():
+				return
+			}
+
+			switch n := notification.(type) {
+			case wallet.NewTransaction:
+				// refresh wallets when new transaction is received
+				pg.UpdateAccountBalance()
+				pg.RefreshWindow()
+			case wallet.SyncStatusUpdate:
+				// refresh wallet account and balance on every new block
+				// only if sync is completed.
+				if n.Stage == wallet.BlockAttached && pg.WL.MultiWallet.IsSynced() {
+					pg.UpdateAccountBalance()
+					pg.RefreshWindow()
+				}
+			}
+		}
+	}()
+}
+
+func (pg *WalletPage) UpdateAccountBalance() {
+	wallets := pg.WL.SortedWalletList()
+
+	pg.listLock.Lock()
+	listItems := pg.listItems
+	pg.listLock.Unlock()
+
+	for _, wal := range wallets {
+		accountsResult, err := wal.GetAccountsRaw()
+		if err != nil {
+			continue
+		}
+
+		var totalBalance int64
+		for _, acc := range accountsResult.Acc {
+			totalBalance += acc.TotalBalance
+		}
+
+		for _, item := range listItems {
+			if item.wal.ID == wal.ID {
+				item.totalBalance = dcrutil.Amount(totalBalance).String()
+				item.accounts = accountsResult.Acc
+			}
+		}
```

Comment:
If this updates the item in `pg.listItems`, then you want to keep `pg.listLock` locked till this method completes. If it doesn't, then this looks pointless because `item` isn't used after here.